prime-router/src/main/kotlin/history/azure/DeliveryFacade.kt

```diff
@@ -95,11 +97,17 @@ class DeliveryFacade(
     fun findDetailedDeliveryHistory(
         deliveryId: Long,
     ): DeliveryHistory? {
-        return dbDeliveryAccess.fetchAction(
+        val deliveryHistory = dbDeliveryAccess.fetchAction(
             deliveryId,
             orgName = null,
             DeliveryHistory::class.java
         )
+        val reportId = deliveryHistory?.reportId
+        if (reportId != null) {
+            val ingestionTime = reportService.getRootReport(UUID.fromString(reportId)).createdAt
```

Comment:
@halprin This is incorrect conceptually -- apologies if I forgot to mention this in our last meeting (I had it in my notes but not sure if I articulated it). 

You can not call "getRootReport" in the delivery function because the reportId that ultimately gets passed in may have MORE THAN ONE ROOT REPORT due to ReportStream's batching feature. The areas in the code where "getRootReport" is called today are guaranteed to only have one root report because those areas are called before batching happens (like the Translation function).

So, one possible solution is to implement the following:

- `getRootReports` - returns one or more root reports
- `DeliveryHistory` - existing class has a new property called `originatingReports` or something that contains the following properties: `ingestionTime`, `reportId`

Please think through the above and how it would work with your product requirements. Let's discuss on call if needed!

Reply:
Thanks for the feedback, @arnejduranovic!  We addressed it.  Please re-review.  :)